sources/rest_api/config_setup.py

```diff
@@ -349,50 +350,96 @@ def _find_resolved_params(endpoint_config: Endpoint) -> List[ResolvedParam]:
     ]
 
 
-def _handle_response_actions(
-    response: Response, actions: List[ResponseAction]
-) -> Optional[str]:
-    """Handle response actions based on the response and the provided actions."""
-    content = response.text
-
-    for action in actions:
+def _action_type_unless_custom_hook(
+    action_type: Optional[str], custom_hook: Optional[List[Callable[..., Any]]]
+) -> Union[
+    Tuple[str, Optional[List[Callable[..., Any]]]],
+    Tuple[None, List[Callable[..., Any]]],
+]:
+    if custom_hook:
+        return (None, custom_hook)
+    return (action_type, None)
+
+
+def _handle_response_action(
+    response: Response, action: ResponseAction
+) -> Union[
+    Tuple[str, Optional[List[Callable[..., Any]]]],
+    Tuple[None, List[Callable[..., Any]]],
+    Tuple[None, None],
+]:
+    """
+    Checks, based on the response, if the provided action applies.
+    """
+    content: str = response.text
+    status_code = None
+    content_substr = None
+    action_type = None
+    custom_hook = None
```

Comment:
One tiny improvement is to use plural for `custom_hook` here: `custom_hooks`, since better aligns with what the function returns, and what `response_action_hook` is expecting here:

```
(action_type, custom_hooks) = _handle_response_action(response, response_action)
```